code/modules/mob/living/simple_animal/hostile/cockroach.dm

```diff
@@ -117,6 +117,8 @@
 			A.visible_message("<span class='danger'>[A] steps onto [src]'s spike!</span>", "<span class='userdanger'>You step onto [src]'s spike!</span>")
 		else if(HAS_TRAIT(A, TRAIT_PACIFISM))
 			A.visible_message("<span class='notice'>[A] carefully steps over [src].</span>", "<span class='notice'>You carefully step over [src] to avoid hurting it.</span>")
+		else if(HAS_TRAIT(A, TRAIT_PIERCEIMMUNE))
+			A.visible_message("<span class='danger'>[A] steps onto [src]'s spike, but is unfazed!</span>", "<span class='userdanger'>You step onto [src]'s spike, but you're unaffected!</span>")
```

Comment:
Are you sure this fixes it?
There's a check earlier to see if the mob is not pierce immune, so wouldn't that last block be what's run if it is?

Was this tested?

Reply:
Tested on alloy golems, but re-tested just to be sure.
Works just as expected between flight, pierceimmune golems, and pacifists, whole 9 yards. 👍